File: blunder/src/world/world.go
package world

type World struct {
	height int
	width  int
	tab    []string
}

func NewWorld(height int, width int, tab string) *World {
	w := new(World)
	w.height = height
	w.width = width
	for _, val := range tab {
		w.tab = append(w.tab, string(val))
	}
	return w
}

func (w *World) PrintWorld() {
	for i := 0; i < w.height; i++ {
		for j := 0; j < w.width; j++ {
			print(w.tab[i*w.width+j])
		}
		println()
	}
}

func (w *World) GetStartPos() (int, int) {
	for i := 0; i < w.height; i++ {
		for j := 0; j < w.width; j++ {
			if w.tab[i*w.width+j] == "@" {
				return i, j
			}
		}
	}
	return -1, -1
}

func (w *World) GetOtherTeleporter(x int, y int) (int, int) {
	for i := 0; i < w.height; i++ {
		for j := 0; j < w.width; j++ {
			if w.tab[i*w.width+j] == "T" && (i != x || j != y) {
				return i, j
			}
		}
	}
	return -1, -1
}

func (w *World) BreakObstacle(x int, y int) {
	w.tab[x*w.width+y] = " "
}

func (w *World) GetBox(x int, y int) string {
	return w.tab[x*w.width+y]
}

func (w *World) GetHeight() int {
	return w.height
}

func (w *World) GetWidth() int {
	return w.width
}

func (w *World) GetTab() []string {
	return w.tab
}

File: blunder-python/world.py
class World:
    def __init__(self, height, width, world_definition):
        self.height = int(height)
        self.width = int(width)
        self.tab = []
        for c in world_definition:
            self.tab.append(c)

    def print_world(self):
        for i in range(self.height):
            line = ""
            for j in range(self.width):
                line += self.tab[i*self.width+j]
            print(line)

    def get_start_position(self):
        for i in range(self.height):
            for j in range(self.width):
                if self.tab[i*self.width+j] == "@":
                    return i, j
        return -1, -1

    def get_other_teleporter(self, x, y):
        for i in range(self.height):
            for j in range(self.width):
                if self.tab[i*self.width+j] == "T" and (i != x or j != y):
                    return i, j
        return -1, -1

    def break_obstacle(self, x, y):
        self.tab[x*self.width+y] = " "
    
    def get_box(self, x, y):
        return self.tab[x*self.width+y]
    
    def get_height(self):
        return self.height
    
    def get_width(self):
        return self.width

    def get_tab(self):
        return self.tab